exes/mqtt-bridge/inc/spark_plug_interface.hpp

```diff
@@ -52,69 +32,278 @@ using org::eclipse::tahu::protobuf::Payload_Metric;
 template <class config_t, class mqtt_client_t>
 class spark_plug_interface {
 public:
-  explicit spark_plug_interface(asio::io_context& io_ctx);
-
-  auto make_will_payload() -> std::string;
+  explicit spark_plug_interface(asio::io_context& io_ctx, config_t& config)
+      : io_ctx_(io_ctx), config_(config),
+        ncmd_topic_(topic_formatter(
+            { constants::namespace_element, config_.value().group_id, constants::ncmd, config_.value().node_id })),
+        mqtt_will_topic_(topic_formatter(
+            { constants::namespace_element, config_.value().group_id, constants::ndeath, config_.value().node_id })),
+        ndata_topic_(topic_formatter(
+            { constants::namespace_element, config_.value().group_id, constants::ndata, config_.value().node_id })) {
+    const std::string_view mqtt_will_payload{ make_will_payload() };
+
+    mqtt_client_ = std::make_unique<mqtt_client_t>(io_ctx_, mqtt_will_topic_, mqtt_will_payload, config_);
+  }
+
+  auto make_will_payload() -> std::string {
+    /// Construct MQTT will payload
+    Payload payload;
+    payload.set_timestamp(timestamp_milliseconds().count());
+
+    auto* metric = payload.add_metrics();
+
+    metric->set_name("bdSeq");
+    metric->set_timestamp(timestamp_milliseconds().count());
+    metric->set_datatype(8);
```

Comment:
what does 8 mean?

Reply:
8 is uint64 data type, I can add comment to clarify

Reply:
does there not exist an enum, which can be used?

Reply:
yes, updating to enum